### PR TITLE
[WiP] removing initialize code from Manager class.

### PR DESCRIPTION
**Pull request by jadelkhoury**
This is still work in progress.

We are aiming remove all initialize code from the Manager class. Instead it should be in the ServletListener Class, or ApplicationBinder.

Issues/questions that remain:
- [ ] How to deal with pagedTRS?
- [ ] storePool and pagedTRS should be instantiated once. Singletons?
- [x] the StorePoolFactory & TRSFactory are not being called. why not?
- [x] ResourcesFactory also needs to be instantiated.

### CODE REVIEW

**Comment by jadelkhoury** [comment]
@berezovskyi ! I made a start but we need to discuss so we can make this correctly, please?

File: adaptor-rm-webapp/src/main/java/com/sample/rm/RMToolManager.java [hunk 1]
```diff
@@ -62,26 +77,17 @@
 
     private static final Logger log = LoggerFactory.getLogger(RMToolManager.class);
 
+    //TODO: Can we inject into a static???
+    @Inject static TrsEventHandler trsEventHandler;
```
Comment (berezovskyi) [inline adaptor-rm-webapp/src/main/java/com/sample/rm/RMToolManager.java]: See `bindAsContract()` in https://stackoverflow.com/questions/49327986/getting-hk2-and-jersey-to-inject-classes

File: adaptor-rm-webapp/src/main/java/com/sample/rm/servlet/ApplicationBinder.java [hunk 2]
```diff
@@ -50,17 +62,70 @@ public ApplicationBinder()
     protected void configure() {
         log.info("HK2 contract binding start");
     
-        bindFactory(new TRSFactory()).to(PagedTrs.class);
-    }
+        bindFactory(new TrsEventHandlerFactory()).to(TrsEventHandler.class);
```
Comment (berezovskyi) [inline adaptor-rm-webapp/src/main/java/com/sample/rm/servlet/ApplicationBinder.java]: add `.in(Singleton.class)`

**Comment by berezovskyi** [comment]
ResourcesFactory also needs to be instantiated.

**Comment by berezovskyi** [comment]
Work notes: `com.sample.rm.RMToolManager#getServiceProviderInfos` cannot be made non-static unless we make `ServiceProviderCatalogSingleton` injectable as well.

**Comment by berezovskyi** [comment]
The TRS base request (http://localhost:8083/adaptor-rm/services/trs/base) fails with the following exceptions (after creating a REQ via http://localhost:8083/adaptor-rm/com/sample/rm/creationdialogsampleclient.jsp?creationUri=http%3A%2F%2Flocalhost%3A8083%2Fadaptor-rm%2Fservices%2Frequirements%2Fcreator):

```
Caused by: org.glassfish.hk2.api.MultiException: A MultiException has 3 exceptions.  They are:
1. org.glassfish.hk2.api.UnsatisfiedDependencyException: There was no object available for injection at SystemInjecteeImpl(requiredType=PagedTrs,parent=TrackedResourceSetService,qualifiers={},position=0,optional=false,self=false,unqualified=null,1005397392)
2. java.lang.IllegalArgumentException: While attempting to resolve the dependencies of org.eclipse.lyo.oslc4j.trs.server.service.TrackedResourceSetService errors were found
3. java.lang.IllegalStateException: Unable to perform operation: resolve on org.eclipse.lyo.oslc4j.trs.server.service.TrackedResourceSetService
```

Upd: fixed in 8ad660d

File: adaptor-rm-webapp/src/main/java/com/sample/rm/servlet/ApplicationBinder.java [hunk 3]
```diff
@@ -66,9 +66,11 @@ protected void configure() {
     
         bindAsContract(RMToolManager.class).in(Singleton.class);
         
-        bindFactory(new TrsEventHandlerFactory()).to(TrsEventHandler.class);
```
Comment (berezovskyi) [inline adaptor-rm-webapp/src/main/java/com/sample/rm/servlet/ApplicationBinder.java]: TrsEventHandlerFactory can now be removed

**Pull request closed without merging**